services/git.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const sg = require('simple-git')()
+const { InvalidResponse, NotFound } = require('.')
+
+async function getDefaultBranch({
+  baseUrl,
+  user,
+  repo,
+  username = '',
+  password = '',
+  simpleGit = sg,
+}) {
+  /*
+  Note on authentication: if we pass the function a valid
+  username and password, they will be used:
+  https://username:password@github.com/badges/shields.git
+  This could be useful where we've got
+  BITBUCKET_USER / BITBUCKET_PASS set, for example.
+
```

Comment:
Would it be feasible to use the auth header instead? I know the userinfo approach is still used, but feel it's generally fallen out of favor (https://tools.ietf.org/html/rfc3986#section-3.2.1)

not familiar with `simple-git` so not sure if/how, but something that results in
```
git -c http.extraheader="Authorization: Basic base64(username:password)"
```

Reply:
We can pass any flags you want, so instead of

```js
console_text = await simpleGit.listRemote(['--symref', url, 'HEAD'])
```

we can call

```js
console_text = await simpleGit.listRemote(['--symref', url, 'HEAD', '-c', 'http.extraheader="Authorization: Basic Og=="'])
```

The issue with this is if it fails, it prompts, which was the exact thing I was trying to avoid. Passing the creds in the URL does prevent that prompting. However, this comment has motivated me to revisit the idea of whether there is a better way to do this and I found `GIT_TERMINAL_PROMPT=0`. The tradeoff is we lose a bit of granularity on the errors because all we know is we couldn't prompt for credentials. I think this is a better/safer option on balance though, and it allows us to rewrite this function so it doesn't require twice as much comment as code to explain what is going on :)

Reply:
> Passing the creds in the URL does prevent that prompting.

That seems like pretty odd behavior. I would've expected providing the same empty/bogus auth value in the header as in the uri userinfo component to have the same behavior :man_shrugging: Anyway, glad it's resolved :+1: